apps/web/src/components/modals/UserSettingsModal.tsx

```diff
@@ -468,6 +470,69 @@ function UserSettings({ user, dispatch, closeModal }: ReduxProps & ModalProps) {
     setConfirmPassword("");
   };
 
+  const updateURLLength = async (item: DropdownItem<"short" | "long" | "normal">) => {
+    const { value } = item;
+    const settings = user.settings;
+
+    const updates: {
+      short_urls: boolean | undefined;
+      long_urls: boolean | undefined;
+    } = {
+      short_urls: undefined,
+      long_urls: undefined,
+    };
+
+    // set to short
+    if (value === "short") {
+      updates.short_urls = true;
+
+      // fix legacy settings
+      if (settings.long_urls) {
+        updates.long_urls = false;
+      }
+      // set to 8 characters
+    } else if (value === "normal") {
+      if (settings.short_urls) {
+        updates.short_urls = false;
+      }
+
+      if (settings.long_urls) {
+        updates.long_urls = false;
+      }
+      // set to long
+    } else if (value === "long") {
+      updates.long_urls = true;
+
+      // fix legacy settings
+      if (settings.short_urls) {
+        updates.short_urls = false;
+      }
+    }
+
+    const { success, error, data } = await makeRequest("PATCH", "/users/@me", {
+      settings: { ...updates },
+    });
```

Comment:
```suggestion
    const { success, error, data } = await makeRequest("PATCH", "/users/@me", {
      settings: { 
        long_urls: value === "long",
        short_urls value === "short",
      },
    });
```

Would this not work?

Reply:
It would but I just didn't want to unnecessarily update data that didn't need to be updated. 🤷‍♀️ 

If you care about the code being clean I wouldn't mind but a better fix would to be allowing multiple fields to be set by patchUser so we don't need to do that garbage and to consolidate our update logic all into patchUser.

Reply:
> It would but I just didn't want to unnecessarily update data that didn't need to be updated. 🤷‍♀️
> 
> If you care about the code being clean I wouldn't mind but a better fix would to be allowing multiple fields to be set by patchUser so we don't need to do that garbage and to consolidate our update logic all into patchUser.

agreed, probably have an overloaded function signature and do a little `typeof` magic...

```ts
function add(a:string, b:string):string;

function add(a:number, b:number): number;

function add(a: any, b:any): any {
    let payload;
    if(typeof a === "number") {} else {}
}
```
^ just stripped that from the [typescript docs ](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads)so the types aren't correct, but im sure it gets the point acrossed.

Reply:
I think either way we're going to run back into the same issue as the original way it was handled where we have to push data to an object then give that to our function.

Unless you have a better solution because I'm assuming you want to give a object param to pass to makeRequest

Reply:
> I think either way we're going to run back into the same issue as the original way it was handled where we have to push data to an object then give that to our function.
> 
> Unless you have a better solution because I'm assuming you want to give a object param to pass to makeRequest

okay okay, what if we just accept the parameters as the object its self.

like:

```ts
patchUser({
   "short_urls": true,
    ...other settings if needed
})
```

and then we have a new function like:
```ts
patchSetting("short_urls", true)
```

and the patchSetting would be a wrapper of patchUser that just passes it as an object instead of two separate params.

unless you have a better idea.

Reply:
Also I did a bit of testing, it seems function overloading is not possible in this context.

Reply:
> Also I did a bit of testing, it seems function overloading is not possible in this context.

https://www.tiktok.com/t/ZTRopo66G/